src/server.js

```diff
@@ -25,6 +25,7 @@ Server.prototype.listen = function(port, host) {
     var client = new Client(true);
     client._end = client.end;
     client.end = function end(endReason) {
+      endReason='{"text":"'+endReason+'"}';
```

Comment:
But are you sure 1.8 just sends "reason" ? Because not only does wiki.vg clearly state it needs to be a "chat", but last time I tried to use .end in 1.8, nmp failed and only printed "Disconnected" on the client side ^^'.